clippy_lints/src/types.rs

```diff
@@ -1107,18 +1106,12 @@ fn numeric_cast_precast_bounds<'a>(cx: &LateContext, expr: &'a Expr) -> Option<(
 
 fn node_as_const_fullint(cx: &LateContext, expr: &Expr) -> Option<FullInt> {
     use rustc::middle::const_val::ConstVal::*;
-    use rustc_const_eval::EvalHint::ExprTypeChecked;
     use rustc_const_eval::ConstContext;
-    use rustc_const_math::ConstInt;
 
-    match ConstContext::with_tables(cx.tcx, cx.tables).eval(expr, ExprTypeChecked) {
+    match ConstContext::with_tables(cx.tcx, cx.tables).eval(expr) {
         Ok(val) => {
             if let Integral(const_int) = val {
-                Some(match const_int.erase_type() {
-                    ConstInt::InferSigned(x) => FullInt::S(x as i128),
-                    ConstInt::Infer(x) => FullInt::U(x as u128),
-                    _ => unreachable!(),
-                })
+                Some(FullInt::U(const_int.to_u128_unchecked()))
```

Comment:
needs a check for the type, to return a `FullInt::S` if the type of `expr` is signed.